src_py/sprite.py

```diff
@@ -1040,131 +1066,137 @@ def draw(self, surface, bgd=None):
-                                       spr.blendmode)
-                else: # dirty sprite
-                    if spr._visible:
-                        _old_rect[spr] = _surf_blit(spr.image,
-                                                    spr.rect,
-                                                    spr.source_rect,
-                                                    spr.blendmode)
-                    if spr.dirty == 1:
-                        spr.dirty = 0
-            _ret = list(_update)
-        else: # flip, full screen mode
+            self._draw_dirty_internal(self.spritedict, _rect,
+                                      self._spritelist,
+                                      _surf_blit, self.lostsprites)
+            _ret = list(self.lostsprites)
+        else:  # flip, full screen mode
             if _bgd is not None:
                 _surf_blit(_bgd, (0, 0))
-            for spr in _sprites:
-                if spr._visible:
-                    _old_rect[spr] = _surf_blit(spr.image,
-                                                spr.rect,
-                                                spr.source_rect,
-                                                spr.blendmode)
-            _ret = [_rect(_clip)] # return only the part of the screen changed
-
+            for spr in self._spritelist:
+                if spr.visible:
+                    self.spritedict[spr] = _surf_blit(spr.image,
+                                                      spr.rect,
+                                                      spr.source_rect,
+                                                      spr.blendmode)
+            _ret = [_rect(_clip)]  # return only the part of the screen changed
 
         # timing for switching modes
         # How may a good threshold be found? It depends on the hardware.
-        end_time = get_ticks()
+        end_time = pygame.time.get_ticks()
         if end_time-start_time > self._time_threshold:
             self._use_update = False
         else:
             self._use_update = True
 
-##        # debug
-##        print "               check: using dirty rects:", self._use_update
-
         # emtpy dirty rects list
-        _update[:] = []
+        self.lostsprites[:] = []
 
         # -------
         # restore original clip
-        _surf.set_clip(_orig_clip)
+        surface.set_clip(_orig_clip)
         return _ret
 
+    @staticmethod
+    def _draw_dirty_internal(_old_rect, _rect, _sprites, _surf_blit,
+                             _update):
+        for spr in _sprites:
+            if spr.dirty < 1:
+                if spr.visible:
+                    # sprite not dirty; blit only the intersecting part
+                    if spr.source_rect is not None:
+                        # For possible future speed up, source_rect's data
+                        # can be pre-fetched outside of this loop.
+                        _spr_rect = _rect(spr.rect.topleft,
```

Comment:
Why are these variables with `_` in front?

Reply:
That seems to have been the coding style of whoever originally wrote the LayeredDirty class, it's just been preserved in my refactor. I don't think the leading underscore has any meaning for temporary variables in a function.